apps/server/src/modules/sharing/sharing.module.ts

```diff
@@ -1,24 +1,25 @@
 import { Module } from '@nestjs/common';
-import { CourseRepo } from '@shared/repo';
+import { CourseRepo, LessonRepo } from '@shared/repo';
 import { LoggerModule } from '@src/core/logger';
 import { AuthorizationModule } from '@src/modules/authorization';
 import { LearnroomModule } from '@src/modules/learnroom';
 import { ShareTokenController } from './controller/share-token.controller';
 import { ShareTokenUC } from './uc';
 import { ShareTokenService, TokenGenerator } from './service';
 import { ShareTokenRepo } from './repo/share-token.repo';
+import { LessonModule } from '../lesson';
 
 @Module({
-	imports: [AuthorizationModule, LearnroomModule, LoggerModule],
+	imports: [AuthorizationModule, LearnroomModule, LessonModule, LoggerModule],
 	controllers: [],
-	providers: [ShareTokenService, TokenGenerator, ShareTokenRepo, CourseRepo],
+	providers: [ShareTokenService, TokenGenerator, ShareTokenRepo, CourseRepo, LessonRepo],
 	exports: [ShareTokenService],
 })
 export class SharingModule {}
 
 @Module({
-	imports: [SharingModule, AuthorizationModule, LearnroomModule, LoggerModule],
+	imports: [SharingModule, AuthorizationModule, LearnroomModule, LessonModule, LoggerModule],
 	controllers: [ShareTokenController],
-	providers: [ShareTokenUC],
+	providers: [ShareTokenUC, CourseRepo, LessonRepo],
```

Comment:
LessonRepo and CourseRepo removed.
- First point: done 
- Second point: I changed to ids, but still can't remove lessonRepo or courseRepo from them
- Third point: Done

Reply:
Update for third point:
CourseService/CourseRepo is required in lessonCopyService because destinationCourseId may not be equal to lesson.course.id. I had to add CourseService and CourseService and CourseRepo there
https://github.com/hpi-schul-cloud/schulcloud-server/blob/b54ff6c0ecd2eb23d533490f884d4de2be5f1a59/apps/server/src/modules/lesson/lesson.module.ts#L11-L26
as when i tried to import LearnroomModule it shows dependency cycle problem. If you have any simple solution to this please let me know, but I think it would require moving a copy of the lesson to a new module, so probably moving it to another Jira ticket would be the best sollution to not make too many changes in one PR.

Reply:
I tried removing this dependency cycle and ended up with 26 files changed (+ more if I wanted to fix the tests) (https://github.com/hpi-schul-cloud/schulcloud-server/commit/794f0cb74f137fed536866ff898015958c87c261). Moreover, I am not familiar with all this architecture, so it would be better if someone with more experience did it (maybe there is a simpler and better way to do it).